src/components/AttachmentCarousel/index.js

```diff
@@ -0,0 +1,204 @@
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'lodash';
+import * as Expensicons from '../Icon/Expensicons';
+import styles from '../../styles/styles';
+import themeColors from '../../styles/themes/default';
+import CarouselActions from './CarouselActions';
+import Button from '../Button';
+import AttachmentView from '../AttachmentView';
+import addEncryptedAuthTokenToURL from '../../libs/addEncryptedAuthTokenToURL';
+import reportActionPropTypes from '../../pages/home/report/reportActionPropTypes';
+import canUseTouchScreen from '../../libs/canUseTouchscreen';
+import CONFIG from '../../CONFIG';
+import CONST from '../../CONST';
+import ONYXKEYS from '../../ONYXKEYS';
+
+const propTypes = {
+    /** sourceUrl is used to determine the starting index in the array of attachments */
+    sourceURL: PropTypes.string,
+
+    /** Object of report actions for this report */
+    reportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
+
+    /** Callback to update the parent modal's state with a sourceUrl and name from the attachments array */
+    onNavigate: PropTypes.func,
+};
+
+const defaultProps = {
+    sourceURL: '',
+    reportActions: {},
+    onNavigate: () => {},
+};
+
+class AttachmentCarousel extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.canUseTouchScreen = canUseTouchScreen();
+        this.makeAndSetArrowState = this.makeAndSetArrowState.bind(this);
+        this.cycleThroughAttachments = this.cycleThroughAttachments.bind(this);
+        this.onShowArrow = this.onShowArrow.bind(this);
+
+        this.state = {showArrows: this.canUseTouchScreen};
+    }
+
+    componentDidMount() {
+        this.makeAndSetArrowState();
+    }
+
+    componentDidUpdate(prevProps) {
+        const prevReportSize = _.size(prevProps.reportActions);
+        const currReportSize = _.size(this.props.reportActions);
+        if (prevReportSize === currReportSize) {
+            return;
+        }
+        this.makeAndSetArrowState();
+    }
+
+    /**
+     * Toggles the visibility of the arrows
+     * @param {Boolean} showArrows
+     */
+    onShowArrow(showArrows) {
+        this.setState({showArrows});
+    }
+
+    /**
+     * Helps to navigate between next/previous attachments
+     * @param {Object} attachmentItem
+     * @returns {Object}
+     */
+    getAttachment(attachmentItem) {
+        const sourceURL = _.get(attachmentItem, 'sourceURL', '');
+        const file = _.get(attachmentItem, 'file', {name: ''});
+        return {
+            sourceURL: addEncryptedAuthTokenToURL(sourceURL),
+            file,
+        };
+    }
+
+    /**
+     * this will be called to set up an attachment array and finding an index
+     */
+    makeAndSetArrowState() {
+        let page;
+        const actionsArr = _.values(this.props.reportActions);
+        const attachments = _.reduce(actionsArr, (attachmentsAccumulator, reportAction) => {
```

Comment:
Function names & comments could be improved. Can you explain what it does?

Reply:
This is to turn `reportComments` into state for this component. This should be called when `reportComments` is updated too.